dockers/for_rocm/Dockerfile

```diff
@@ -0,0 +1,26 @@
+
+FROM rocm/pytorch:latest
+
+RUN \
+    pip install click && \
+    pip install deepdiff && \
+    pip install fastapi && \
+    pip install lightning-cloud && \
+    pip install -U Jinja2 && \
+    pip install starsessions && \ 
+    pip uninstall fsspec && \ 
+    pip install fsspec==2022.1.0 && \ 
+    pip install jsonargparse && \
+    pip install -U jsonargparse[signatures]
+
+#for flash 
+RUN python -m pip install urllib3
+
+#for bolts and transformers
+RUN pip install protobuf==3.9.2
+
+RUN \
```

Comment:
Can we actually keep this similar to https://github.com/Lightning-AI/lightning/blob/master/dockers/base-cuda/Dockerfile ? So that they are virtually the same except for rocm vs cuda?

Reply:
> Also when we do add the dockerfile here, we at least need a CI building that image :)

As to the CI building, we're looking if we can provide a machine for that but I am unsure when I can get a machine for this. Do you have any github action resources free to build the docker image?

Reply:
> Can we actually keep this similar to https://github.com/Lightning-AI/lightning/blob/master/dockers/base-cuda/Dockerfile ? So that they are virtually the same except for rocm vs cuda?

We have updated the base-rocm file to be as similar to the base-cuda file as possible.

Reply:
> Do you have any github action resources free to build the docker image?

@sgschwindAMD I tried pushing a commit to your branch, but it seems like you don't allow us to push any commits. Please cherry-pick this commit: https://github.com/Lightning-AI/lightning/commit/756692335d8a8d21c45fc58738dc31d246d35f59